bugfixes, cleanups and improvements

xavierba opened the pull request:
Hi Soji,

Here's a pile of patches for you to review. I expect most of them to be fine as is, but there is a non null probability I'm missing some background and I'm indeed open to discussion :-)
I have one more patch pending in my WIP branch, which changes the location and handling of  sympa aliases. As such, it is a bit more disruptive and needs to be discussed.

Regards,
Xavier

xavierba committed:
sympa 6.2.19 beta 2

xavierba committed:
Fix create_cookie to only replace the cookie config and not random comments in the conf file

xavierba committed:
Move README to doc dir

xavierba committed:
Don't hide intranet choices

xavierba committed:
Fix creation of directory structure for overiding default configs

xavierba committed:
Remove function to migrate legacy /etc/sysconfig/sympa

xavierba committed:
Remove un-necessary owner/group fixup of main and robot confs

xavierba committed:
Move rsyslog conf installation next to logrotate conf installation

xavierba committed:
Re-order %files

xavierba committed:
Remove useless ca-bundle.crt

xavierba committed:
Use %make_build and %make_install

xavierba committed:
Don't override PATH

xavierba committed:
Rename README.RPM.md

xavierba committed:
Add note to unbundle fonts

xavierba committed:
Don't ship bogus smtpc.1.md

xavierba committed:
Fix typos in README.RPM.md

xavierba committed:
Add changelog entry

xavierba committed:
Use mktemp to create temporary cookie file in %post

xavierba committed:
Ignore test suite results on F25 and F26, earlier and later releases are fine

xavierba committed:
Replace deprecated fcgid directives with their new names

xavierba committed:
FcgidMaxProcesses should be set in /etc/httpd/conf.d/fcgid.conf if needed, but not in sympa.conf

xavierba committed:
Rename httpd conf according to httpd version

xavierba committed:
data_structure.current_version needs to be owned by user and group sympa

xavierba committed:
doc: Setup MTA before DB to receive mail sent by 'sympa.pl --health_check'

ikedas commented on SPECS/sympa-6.2.spec.in:
```diff
@@ -335,6 +335,7 @@ cp -p %{SOURCE113} __doc/README.RPM.md
 %else
 cp -p %{SOURCE112} __doc/README.RPM.md
 %endif
+mv %{buildroot}%{_sysconfdir}/sympa/README __doc/
```
_CONFDIR_/README (etc_README in source) describes usage of _CONFDIR_ and _DEFAULTDIR_. So it would be on either of these directories.

xavierba replied:
I understand the rationale for putting this in either (or both) of configdir and defaultdir, but documentation definitely belongs to docdir. Maybe I could move the file to docdir, but also symlink it from both configdir and defaultdir ?

ikedas commented on SPECS/sympa-6.2.spec.in:
```diff
@@ -291,6 +291,9 @@ pushd po/web_help; rm -f stamp-po; make; popd
 # Fix perm to prevent warning by rpmlint.
 chmod a-x %{buildroot}%{_datadir}/%{name}/bin/create_db.Sybase
 
+# Remove useless ca-bundle.crt
```
This file is referred by the code (see [Conf.pm](https://github.com/sympa-community/sympa/blob/6.2.22/src/lib/Conf.pm#L1923)).  It would be better to propose upstream removal.

xavierba replied:
Yes, I noticed the issue and worked around it by adding a symlink.
I've filled an issue about this upstream, as you suggested.
https://github.com/sympa-community/sympa/issues/116

ikedas commented on SPECS/sympa-6.2.spec.in:
```diff
@@ -342,6 +342,8 @@ cp -p %{SOURCE113} __doc/README.RPM.md
 %endif
 mv %{buildroot}%{_sysconfdir}/sympa/README __doc/
+# Remove smtpc.1.md (this should be made a manpage instead)
+rm __doc/smtpc.1.md
```
There is no manpage: smtpc.1.md is only documentation.

xavierba replied:
Ok, I will drop this commit.

ikedas commented on SPECS/sympa-6.2.spec.in:
```diff
@@ -425,24 +425,6 @@ function create_config {
-            if [ '!' -e %{_sysconfdir}/sympa/scenari/$scenario ]; then
-                touch %{_sysconfdir}/sympa/scenari/$scenario:ignore
-            fi
-        done
```
intranet settings are incomplete examples and it would be better to disable them by default.

xavierba replied:
I guess the removal should be done upstream, I've filed an issue about this : 
https://github.com/sympa-community/sympa/issues/119
Meanwhile, I'll also remove the intranet related files (default/create_list_templates/intranet_list and default/scenari/*intranet*) in %install, which should have the same effect than the removed code.
(it would be better to remove the files in %prep, but then configure complains, as the Makefile needs to be tweaked too)

ikedas commented on SPECS/sympa-6.2.spec.in:
```diff
@@ -547,17 +562,26 @@ fi
 %license COPYING
 %dir %attr(-,sympa,sympa) %{_sysconfdir}/sympa/
 %config(noreplace) %attr(0640,sympa,sympa) %{_sysconfdir}/sympa/sympa.conf
-%config(missingok) %{_sysconfdir}/sympa/create_list_templates
```
Files under _sysconfdir (except `sympa.conf`) are optional: Files under DIFAULTDIR are used by default.
And any or most of these files can be removed or moved by upstream in the future.
So I suppose `missingok` is better than `noreplace`.

xavierba replied:
Having all the config files available make it easier to understand how to tune sympa behaviour, so I'd like to keep them. But as you outlined, these config files are actually not mandatory, so I'll go with %config(noreplace,missingok) which should allow for both. If file are removed or moved upstream, it would need to be acommodated for in the specfile anyway, so this shouldn't be an issue.

ikedas commented on SOURCES/sympa-httpd-fcgid-el7.conf:
```diff
@@ -2,10 +2,10 @@
-    MaxProcessCount 5
-    MaxRequestLen 131072
+    FcgidMaxProcesses 5
+    FcgidMaxRequestLen 131072
```
See a comment on #9.

xavierba replied:
The rationale for keeping FcgidIOTimeout and FcgidMaxRequestLen but discard FcgidMaxProcesses is the formers are valid inside a VirtualHost while the later is not.
I'll add another patch to plain remove them and add a comment.

ikedas commented on SOURCES/sympa-systemd-README.RPM.md:
```diff
@@ -13,7 +13,107 @@ package.  A bunch of work is needed to start your Sympa service.
     Sympa and updates site configuration files.  In this process you must
     choose e-mail(s) of listmaster, database settings and so on.
 
-2 Setup database
```
Configuration can refer database.  So I suppose database setup would be better to be done at first.

xavierba replied:
My point is if one follows the steps in README.RPM closely and setup the database before setting up the DB, he will not get the mail to sympa@domain.tld sent by 'sympa.pl --health_check', thus the MTA needs to be setup before the DB.

ikedas reviewed (commented):
@xavierba, sorry for delayed response. I've submitted comments.

xavierba commented:
I've created a new branch PR-20171108 hopefully addressing your comments and some more fixup, but I don't know how to update the pull request to target the new branch. Sorry, not used to github workflow, I'm probably not doing that in the right way at all...

xavierba commented:
I've created a new branch PR-20171108 hopefully addressing your comments and some more fixup, but I don't know how to update the pull request to target the new branch. Sorry, not used to github workflow, I'm probably not doing that in the right way at all...

xavierba commented:
This PR is replaced by https://github.com/ikedas/sympa-rpm/pull/13
